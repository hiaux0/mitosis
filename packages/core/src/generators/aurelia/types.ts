import { OmitObj } from '../../helpers/typescript';
import { BaseTranspilerOptions } from '../../types/transpiler';
import { MitosisImport } from '../../types/mitosis-component';

export type AureliaVersion = 1 | 2;
export const AureliaV1 = 1;
export const AureliaV2 = 2;

interface AureliaVersionOpt {
  aureliaVersion: AureliaVersion;
}

export interface ToAureliaOptions extends BaseTranspilerOptions, AureliaVersionOpt {
  importMapper?: Function;
  // cssNamespace?: () => string;
  // namePrefix?: (path: string) => string;
  // asyncComponentImports?: boolean;
  // api: Api;
}

export type AureliaOptsWithoutVersion = OmitObj<ToAureliaOptions, AureliaVersionOpt>;

export interface ImportValues {
  starImport: string | null;
  defaultImport: string | null;
  namedImports: string | null;
}

export interface ImportData {
  name: string;
  path: string;
  templatePath: string;
  jsPath: string;
  imports: MitosisImport['imports'];
}
