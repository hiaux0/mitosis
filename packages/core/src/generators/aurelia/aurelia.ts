import { componentToAurelia } from './generate';
import { AureliaOptsWithoutVersion } from './types';

export const componentToAurelia1 = (aureliaOptions?: AureliaOptsWithoutVersion) =>
  componentToAurelia({ ...aureliaOptions, aureliaVersion: 1 });

export const componentToAurelia2 = (aureliaOptions?: AureliaOptsWithoutVersion) =>
  componentToAurelia({ ...aureliaOptions, aureliaVersion: 2 });
