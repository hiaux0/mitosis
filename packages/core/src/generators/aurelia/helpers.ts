import { stripStateAndPropsRefs } from '../../helpers/strip-state-and-props-refs';
import { MitosisComponent } from '../../types/mitosis-component';
import { ToAureliaOptions } from './types';

export const stripStateAndProps =
  ({ options, json }: { options: ToAureliaOptions; json: MitosisComponent }) =>
  (code: string) =>
    stripStateAndPropsRefs(code, {
      replaceWith: (name) => {
        const result = name === 'children' ? '$$slots.default' : `this.${name}`;
        return result;
      },
    });
