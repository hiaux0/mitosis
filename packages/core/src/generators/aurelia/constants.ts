import { MitosisComponent, MitosisImport } from 'src/types/mitosis-component';
import { getImportValue } from '../../helpers/render-imports';
import { AureliaVersion, ImportValues, ToAureliaOptions } from './types';

export const DEFAULT_AURELIA_VERSION: AureliaVersion = 1;
export const IMPORT_MARKER = '[[MARKER]]';
export const MARKER_JS_MAPPED = '[[JS_MARKER]]';

export const DEFAULT_AURELIA_OPTIONS: ToAureliaOptions = {
  aureliaVersion: DEFAULT_AURELIA_VERSION,

  importMapper: (
    // ...args
    component: MitosisComponent | null | undefined,
    theImport: MitosisImport,
    importedValues: ImportValues,
    componentsUsed: string[],
    path: string,
  ) => {
    const importValue = getImportValue(importedValues);

    const templateMapped = importValue
      ? `${IMPORT_MARKER}<require from="${path}"></require>`
      : `import '${path}';`;

    const jsMapped = importValue ? `import ${importValue} from '${path}';` : `import '${path}';`;

    const finalMapped = `${templateMapped}${MARKER_JS_MAPPED}${jsMapped}`;

    return finalMapped;
  },
};
