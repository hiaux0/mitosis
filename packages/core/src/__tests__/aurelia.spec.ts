import { DEFAULT_AURELIA_OPTIONS } from '../generators/aurelia/constants';
import { componentToAurelia } from '../generators/aurelia';
import { runTestsForTarget } from './shared';

describe('Aurelia', () => {
  runTestsForTarget({
    options: DEFAULT_AURELIA_OPTIONS,
    target: 'aurelia',
    generator: componentToAurelia,
  });
  // runTestsForTarget({
  //   options: {
  //     standalone: true,
  //   },
  //   target: 'aurelia',
  //   generator: componentToAurelia,
  // });
});
