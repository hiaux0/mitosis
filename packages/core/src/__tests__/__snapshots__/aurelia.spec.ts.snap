// Vitest Snapshot v1

exports[`Aurelia > jsx > Typescript Test > Basic 1`] = `
"import { NgModule } from \\"@aurelia/core\\";
import { CommonModule } from \\"@aurelia/common\\";

import { Component } from \\"@aurelia/core\\";

export interface MyBasicComponentProps {
  id: string;
}

export const DEFAULT_VALUES = {
  name: \\"Steve\\",
};

@inlineView({
  selector: \\"my-basic-component, MyBasicComponent\\",
  template: \`
       <div class=\\"test div\\">
         <input
           [attr.value]=\\"DEFAULT_VALUES.name || name\\"
           (input)=\\"name = $event.target.value\\"
         />

         Hello! I can run in React, Vue, Solid, or Liquid!
       </div>

       \`,
  styles: [
    \`.div {
         padding: 10px;
       }
\`,
  ],
})
export class MyBasicComponent {
  DEFAULT_VALUES = DEFAULT_VALUES;

  name = \\"Steve\\";
  underscore_fn_name() {
    return \\"bar\\";
  }
  age = 1;
  sports = [\\"\\"];
}

@NgModule({
  declarations: [MyBasicComponent],
  imports: [CommonModule],
  exports: [MyBasicComponent],
})
export class MyBasicComponentModule {}
"
`;

exports[`Aurelia > svelte > Typescript Test > ifElse 1`] = `
"import { NgModule } from \\"@aurelia/core\\";
import { CommonModule } from \\"@aurelia/common\\";

import { Component } from \\"@aurelia/core\\";

@inlineView({
  selector: \\"my-component, MyComponent\\",
  template: \`
       <ng-container *ngIf=\\"show\\">
         <button (click)=\\"toggle($event)\\">Hide</button>
       </ng-container>

       \`,
})
export class MyComponent {
  show = true;
  toggle = function toggle() {
    this.show = !this.show;
  };
}

@NgModule({
  declarations: [MyComponent],
  imports: [CommonModule],
  exports: [MyComponent],
})
export class MyComponentModule {}
"
`;
